### PR TITLE
fix: bootstrap contract for GPUs to include GPUImageSHA

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
**Description**
Agentbaker [PR](https://github.com/Azure/AgentBaker/pull/3849/files) adds an additional field to the bootstrap script that removes the hardcoded value for GPU_IMAGE_SHA, and instead requires the Contract Consumer to supply this value for Ubuntu GPU Bootstrapping. 

This Broke bootstrapping for these two image versions, for all the [Ubuntu2204containerd Community Images](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/providers/Microsoft.Compute/locations/westus2/CommunityGalleries/AKSUbuntu-38d80f77-467a-481f-a8d4-09b6d4220bd2/Images/2204gen2containerd/overview)
- 202312.06.0
- 202312.12.0

This PR adds the GPUImageSHA required field to our node bootstrapping variables, which fixes GPU bootstrapping for these two image versions

**How was this change tested?**
- make e2etest to run GPU scenario 
- additional code coverage 
- manual scale up 

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note
fixing GPU bootstrapping for Ubuntu2204containerd
```

### CODE REVIEW

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

File: designs/gpu-selection-and-bootstrap.md [hunk 1]
```diff
@@ -108,10 +109,10 @@ Here are some relevant labels:
 
 | Label                                 | Description                               |
 |---------------------------------------|-------------------------------------------|
-| `karpenter.k8s.azure/sku-family`       | Family of the SKU (e.g., standardAv2Family)|
-| `karpenter.k8s.azure/sku-subfamily`    | Subfamily of the SKU                      |
-| `karpenter.k8s.azure/sku-cpu`          | Number of virtual CPUs                    |
-| `karpenter.k8s.azure/sku-accelerator`  | Type of accelerator (e.g., Nvidia)        |
+| `karpenter.azure.com/sku-family`       | Family of the SKU (e.g., standardAv2Family)|
```
Comment (cameronmeissner) [inline designs/gpu-selection-and-bootstrap.md:112]: unrelated changes?

File: pkg/providers/imagefamily/bootstrap/aksbootstrap.go [hunk 2]
```diff
@@ -202,6 +202,7 @@ type NodeBootstrapVariables struct {
 	KubeletConfigFileEnabled          bool     // s   can be static	[should kubelet config be actually used/preferred instead of flags?]
 	KubeletConfigFileContent          string   // s   mix of user/static/RP-generated.
 	SwapFileSizeMB                    int      // t   user input
+	GPUImageSHA                       string   // s static sha rarely updated
```
Comment (cameronmeissner) [inline pkg/providers/imagefamily/bootstrap/aksbootstrap.go:205]: nit: comment allignment doesn't match

**Review by cameronmeissner — DISMISSED** [review]

File: designs/gpu-selection-and-bootstrap.md (hunk 1 above)
Comment (Bryce-Soghigian) [reply designs/gpu-selection-and-bootstrap.md:112]: yeah, just figured, i should update these gpu docs while im here

**Review by tallaxes — COMMENTED** [review]
/test

File: designs/gpu-selection-and-bootstrap.md (hunk 1 above)
Comment (tallaxes) [reply designs/gpu-selection-and-bootstrap.md:112]: `sku-family` value is just a single letter, e.g. "D"

File: designs/gpu-selection-and-bootstrap.md [hunk 3]
```diff
@@ -108,10 +109,10 @@ Here are some relevant labels:
...
+| `karpenter.azure.com/sku-subfamily`    | Subfamily of the SKU                      |
```
Comment (tallaxes) [inline designs/gpu-selection-and-bootstrap.md:113]: We don't support subfamily

**Review by tallaxes — APPROVED** [review]

File: designs/gpu-selection-and-bootstrap.md (hunk 3 above)
Comment (Bryce-Soghigian) [reply designs/gpu-selection-and-bootstrap.md]: removed, although I think we should have a conversation of supporting it for gpu

File: pkg/providers/imagefamily/bootstrap/aksbootstrap.go (hunk 2 above)
Comment (Bryce-Soghigian) [reply pkg/providers/imagefamily/bootstrap/aksbootstrap.go]: fixed

File: designs/gpu-selection-and-bootstrap.md (hunk 1 above)
Comment (Bryce-Soghigian) [reply designs/gpu-selection-and-bootstrap.md]: good catch

**Review by Bryce-Soghigian — COMMENTED** [review]
/test

**Review by tallaxes — APPROVED** [review]
